src/Tokenizer/Tokens.php

```diff
@@ -879,8 +879,16 @@ public function insertAt($index, $items)
         $this->changed = true;
         $this->setSize($oldSize + $itemsCnt);
 
+        // since we only move already existing items around, we directly call into SplFixedArray::offset* methods.
+        // that way we get around additional overhead this class adds with overridden offset* methods.
         for ($i = $oldSize + $itemsCnt - 1; $i >= $index; --$i) {
-            $this[$i] = $this->offsetExists($i - $itemsCnt) ? $this[$i - $itemsCnt] : new Token('');
+            if (!$this->offsetExists($i - $itemsCnt)) {
+                $this[$i] = new Token('');
+            } else {
+                $oldItem = parent::offsetGet($i - $itemsCnt);
+                $this->blockEndCache = [];
+                parent::offsetSet($i, $oldItem);
```

Comment:
Actually I am not sure why tests are failling. Most important point of this PR is getting rid of this->offsetSet and using parent::offsetSet because there lays the most overhead